File: Core/Models/Vehicle.cs
using System;
using System.Collections.Generic;
using System.Collections.ObjectModel;

namespace AutoCity.Core.Models
{
    public class Vehicle
    {
        public int Id { get; set; }
		public int ModelId { get; set; }
		public Model Model { get; set; }
		public int YearMade { get; set; }
		public DateTime LastUpdate { get; set; }
		public int Price { get; set; }
		public string VinNumber { get; set; }
		public string TransmissionType { get; set; }
		public int Mileage { get; set; }
		public string Color { get; set; }
		public int MPG { get; set; }
		public string Features { get; set; }
		public ICollection<Photo> Photos { get; set; }
		public Vehicle()
		{
			Photos = new Collection<Photo>();
		}
    }
}

File: ClientApp/app/models/vehicle.ts
export interface KeyValuePair{
	id: number,
	name: string
}
export interface Vehicle {
	id: number,
	model: any,
	make: any,
	yearMade: number,
	lastUpdate: string,
	price: number,
	vinNumber: string,
	transmissionType: string,
	mileage: number,
	color: string,
	mpg: number,
	features: string
}

export interface SaveVehicle {
	id: number,
	makeId: number,
	modelId: number,
	yearMade: number,
	price: number,
	vinNumber: string,
	transmissionType: string,
	mileage: number,
	color: string,
	mpg: number,
	features: string
}